content/en/guide/v10/typescript.md

```diff
@@ -216,19 +210,15 @@ export class Button extends Component {
 
 ## Typing references
 
-The `createRef` function is also generic, and lets you bind references to element types. In this example, we ensure that the reference can only be bound to `HTMLAnchorElement`. Using `ref` with any other element lets TypeScript thrown an error:
+`createRef()`is generic type. You can specify a reference type by passing `createRef()`'s generic type parameter:
 
 ```tsx
 import { h, Component, createRef } from "preact";
 
 class Foo extends Component {
+  // `this.ref.current` is HTMLAnchorElement type.
   ref = createRef<HTMLAnchorElement>();
 
-  componentDidMount() {
-    // current is of type HTMLAnchorElement
```

Comment:
`this.ref.current` is `div`.
This comment is confusing.

Reply:
Good catch!